src/Areas/AzureArc/Services/ArcServices.cs

```diff
@@ -0,0 +1,298 @@
+                string tempScriptPath = Path.Combine(userProvidedPath, "RemoveAksEdgeCompletely.ps1");
+                File.WriteAllText(tempScriptPath, LoadResourceFiles(RemoveAksEdgeCompletely));
+
+                var processStartInfo = new ProcessStartInfo
+                {
+                    FileName = "powershell.exe",
+                    Arguments = $"-ExecutionPolicy Bypass -File \"{tempScriptPath}\" -Force",
+                    UseShellExecute = true,
+                    Verb = "runas"
+                };
+
+                using var process = StartProcess(tempScriptPath, processStartInfo);
+                await process.WaitForExitAsync();
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"Error during AKS Edge removal: {ex.Message}");
+            }
+
+            try
+            {
+                // Extract the confirmation script to a temporary file
+                string tempConfirmScriptPath = Path.Combine(userProvidedPath, "ConfirmAksEdgeDeletion.ps1");
+                File.WriteAllText(tempConfirmScriptPath, LoadResourceFiles(ConfirmAksEdgeDeletion));
+
+                var confirmProcessStartInfo = new ProcessStartInfo
+                {
+                    FileName = "powershell.exe",
+                    Arguments = $"-ExecutionPolicy Bypass -File \"{tempConfirmScriptPath}\" -Force",
+                    UseShellExecute = true,
+                    Verb = "runas"
+                };
+
+                using var confirmProcess = StartProcess(tempConfirmScriptPath, confirmProcessStartInfo);
+                await confirmProcess.WaitForExitAsync();
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"Error during confirmation script execution: {ex.Message}");
+            }
+
+            return true;
+        }
+        public string LoadResourceFiles(string resourceName)
+        {
+            string actualResourceName = EmbeddedResourceHelper.FindEmbeddedResource(_assembly, resourceName);
+            return EmbeddedResourceHelper.ReadEmbeddedResource(_assembly, actualResourceName);
+        }
+
+        public async Task<DeploymentResult> ValidatePrerequisitesForAksEdgeClusterAsync()
+        {
+            string prerequisiteSteps = await Task.Run(() => LoadResourceFiles(PrerequisitesAksEdgeInstallation));
+            return new DeploymentResult
+            {
+                Success = true,
+                Steps = prerequisiteSteps
+            };
+        }
+
+        public async Task<DeploymentResult> ValidateSystemRequirementsAndSetupHyperVAsync(string userProvidedPath)
+        {
+            // Ensure the directory exists
+            if (!Directory.Exists(userProvidedPath))
+            {
+                Directory.CreateDirectory(userProvidedPath);
+            }
+
+            string tempScriptPath = Path.Combine(userProvidedPath, "ValidateAndSetupSystemRequirements.ps1");
```

Comment:
I think the pattern you are using is wrapping ps1 script in the services.  To me, this seems like it could be flaky and error prone.  Do we have alternatives?  Like do we have SDKs we can call for this?  Based on current implementation I can see how this could be difficult to debug.